### DISCUSSION TITLE
Connection Pooling to RDS Proxy

### DISCUSSION THREAD

**Discussion by eroltoker** [brefphp/bref#1753] [Support]
Hello, 

I'm using RDS proxy which works really well, but requires the lambda is deployed in the same VPC to be able to access the RDS proxy endpoint.  I've read about how bref recommends PlanetScale but if you're heavily using mysql's native functionality, it's a risk to swap to another service even if it has a same interface (it may not scale the same way, or down the line my needs may change and require mysql native functionality).

I'm running a fairly standard LAMP stack type setup with bref right now.  Are there any creative ideas on how to solve this connection pooling problem?

I've thought of deploying my own proxysql server or bastion host, but this defeats the point of serverless 😭.

Appreciate your inputs.

**Comment by deleugpn** [comment]
You just need to deploy your Lambda in the same VPC as the RDS and the RDS Proxy. What's the issue with that?

**Comment by eroltoker** [comment]
A few things:
1) I have services outside AWS that can't leverage the proxy for connection pooling
2) I'm running some services that benefit from the ephemeral nature of lambda deploying the function to different servers over time (helps with fingerprinting)
3) You have to run a NAT gateway and pay for that separately.

**Reply by mnapoli** [reply]
1: not all services have to use the RDS proxy
2: I'm not sure I understand what you meant
3: to be honest, if you are at a scale that you need RDS proxy, NAT Gateway is usually not so much of a concern.

**Comment by eroltoker** [comment]
I'll close this, adding one more fun thing that I learned... what I pay for the VPC + Nat Gateway usage costs is almost 50% of the cost of my RDS instance 😭😭😭.  I configured proxySQL, so i'll switch to using that.
